src/pcre2_substitute.c

```diff
@@ -875,6 +879,12 @@ do
         goto LITERAL;
 
         default:
+            if (rc < 0)
+              {
+              special = 0;
+              group = -rc;
+              goto GROUP_SUBSTITUTE;
```

Comment:
This sort of goto-based code leaves me terrified. Surely all the common blocks of code should be pulled out into functions instead of gotos! Ouch.

But for a minimal fix, I added another goto. The danger is that we need to be very, very sure that we are jumping past the initialisation of any important stack variables.

I have checked as carefully as I can, that only `special` and `group` need to be set for the jump (and `name` would be required if `group < 0`).

Reply:
The gotos in PCRE2 would mostly be fine by 2024 standards, you're right. Certainly the error-handling ones at the end of the functions are OK in "modern C" (if that's even a thing...).

I'm not judging any existing code, really. It's my own goto that I've just added that frightens me the most, although I am confident, after staring at it for a while.

Tweaked code to silence the warning, which was due to the poor compiler not being able to reason about which variables were initialised before use (due to the new goto!)